### PR TITLE
Issue #199: Modify format of patch body per the Json Patch format.

### PR DESCRIPTION
**Pull request by suratdas**
Implement https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/199 
Does not impact SDK.

### CODE REVIEW

File: src/builds/builds.service.ts [hunk 1]
```diff
@@ -100,7 +99,7 @@ export class BuildsService {
     return new BuildDto(build);
   }
 
-  async update(id: string, modifyBuildDto: ModifyBuildDto): Promise<BuildDto> {
+  async update(id: string, modifyBuildDto: object): Promise<BuildDto> {
```
Comment (pashidlos) [inline src/builds/builds.service.ts:102]: `Prisma.BuildUpdateInput` could be used as type

File: src/builds/builds.controller.ts [hunk 2]
```diff
@@ -68,12 +69,17 @@ export class BuildsController {
...
-  update(@Param('id', new ParseUUIDPipe()) id: string, @Body() modifyBuildDto?: ModifyBuildDto): Promise<BuildDto> {
+  update(@Param('id', new ParseUUIDPipe()) id: string, @Body() patchObject?: object[]): Promise<BuildDto> {
     //In future, no or empty body will do nothing as this check will be removed. It will expect a proper body to perform any patch.
-    if (modifyBuildDto === null || Object.keys(modifyBuildDto).length === 0) {
-      modifyBuildDto.isRunning = false;
+    if (patchObject === null || Object.keys(patchObject).length === 0) {
+      return this.buildsService.update(id, { "isRunning": false });
+    } else {
+      let target = {};
+      (patchObject as Operation[]).forEach(eachOpeartion => {
```
Comment (pashidlos) [inline src/builds/builds.controller.ts:78]: not sure how it's better then initial `ModifyBuildDto`

**Review by pashidlos — CHANGES_REQUESTED** [review]
@suratdas I appreciate your efforts but don't see this change solve any problem 
I like your previous implementation better as it has type checks 

is there any problem that we could address with new implementation?

**Comment by suratdas** [comment]
@pashidlos I was trying to make it per the specification. I know it does not change any functionality. If it does not add too much value, we can completely descope this. Some of the reference materials (including spec) that talk about the proper implementation of PATCH are below.

https://williamdurand.fr/2014/02/14/please-do-not-patch-like-an-idiot/
https://www.baeldung.com/spring-rest-json-patch
https://tools.ietf.org/html/rfc6902
https://apisyouwonthate.com/blog/put-vs-patch-vs-json-patch

**Comment by pashidlos** [comment]
@suratdas sorry, my bad 
as I understand it's the proposed way to update complex json objects
in our case it's handled by [Prisma](https://www.prisma.io/docs/concepts/components/prisma-client/crud#update) api that allows us provide only needed properties to update instead of full document replacement without loosing strict typing 
I'll postpone this change until we hit any restriction from Prisma side and we need external tool to overcome it

**Pull request closed without merging**